internal/cephfs/nodeserver.go

```diff
@@ -127,10 +133,55 @@ func maybeUnlockFileEncryption(
 	stagingTargetPath string,
 	volID fsutil.VolumeID,
 ) error {
-	if volOptions.IsEncrypted() {
-		log.DebugLog(ctx, "cephfs: unlocking fscrypt on volume %q path %s", volID, stagingTargetPath)
 
-		return fscrypt.Unlock(ctx, volOptions.Encryption, stagingTargetPath, string(volID))
+	if volOptions.IsEncrypted() == false {
+		return nil
+	}
+
+	// Define Mutex Lock variables
+	lockName := string(volID) + "-mutexLock"
+	lockCookie := lockName + "-coockie"
+	lockDesc := "Lock for " + string(volID)
+	lockDuration := 150 * time.Second
+	var flags byte = 0
+
+	log.DebugLog(ctx, "Creating lock for the following volume ID %s", volID)
+
+	ioctx, err := volOptions.GetConnection().GetIoctx(volOptions.MetadataPool)
+	if err != nil {
+		log.ErrorLog(ctx, "Failed to create ioctx: %s", err)
+	}
+	defer ioctx.Destroy()
+
+	res, err := ioctx.LockExclusive(volOptions.VolID, lockName, lockCookie, lockDesc, lockDuration, &flags)
+	switch res {
```

Comment:
can we make use of https://pkg.go.dev/syscall? and also need to return errors instead of returning nil until the fscrypt.Unlock is done or how to ensure that unlock is successful?

Reply:
i think we might need to make use of metadata to ensure it's in progress or is done along with the lock.

Reply:
I do not see how the syscall package can be used here.

Reply:
Thank you for the reviews @Madhu-1. 

Sorry, I might misunderstood you.  `fscrypt.Unlock` returns an error if it is not successful. 

Could you please elaborate more regarding https://pkg.go.dev/syscall?

Reply:
i mean like below

```go
if errors.Is(err, syscall.EEXIST) {
```

if we do like above we don't need to import c to this file